src/MIDebugEngine/Engine.Impl/DebuggedThread.cs

```diff
@@ -231,6 +301,72 @@ private ThreadContext CreateContext(TupleValue frame)
 
             return new ThreadContext(pc, textPosition, func, level, from);
         }
+
+        private bool TryGetTidFromTargetId(string targetId, out uint tid)
+        {
+            tid = 0;
+            if (System.UInt32.TryParse(targetId, out tid) && tid != 0)
+            {
+                return true;
+            }
+            else if (targetId.StartsWith("Thread ", StringComparison.OrdinalIgnoreCase) &&
+                     System.UInt32.TryParse(targetId.Substring("Thread ".Length), out tid) &&
+                     tid != 0
+            )
+            {
+                return true;
+            }
+            else if (targetId.StartsWith("Process ", StringComparison.OrdinalIgnoreCase) &&
+                    System.UInt32.TryParse(targetId.Substring("Process ".Length), out tid) &&
+                    tid != 0
+            )
+            {   // First thread in a linux process has tid == pid
+                return true;
+            }
+            else if (targetId.StartsWith("Thread ", StringComparison.OrdinalIgnoreCase))
+            {
+                // In processes with pthreads the thread name is in form: "Thread <0x123456789abc> (LWP <thread-id>)"
+                int lwp_pos = targetId.IndexOf("(LWP ", StringComparison.Ordinal);
+                int paren_pos = targetId.LastIndexOf(')');
+                int len = paren_pos - (lwp_pos + 5);
+                if (len > 0 && System.UInt32.TryParse(targetId.Substring(lwp_pos + 5, len), out tid) && tid != 0)
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                tid = --s_targetId;
+                return true;
+            }
+
+            return false;
+        }
+
+        private DebuggedThread SetThreadInfoFromResultValue(ResultValue resVal, out bool newThread)
+        {
+            newThread = false;
+            int threadId = resVal.FindInt("id");
+            string targetId = resVal.TryFindString("target-id");
+
+            DebuggedThread thread = FindThread(threadId, out newThread);
+            thread.Alive = true;
```

Comment:
@gregg-miskelly This was there already. I just moved it. I think its there because if thread info lists it, it is still alive.